src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UriConverter.cs

```diff
@@ -21,5 +24,16 @@ public override void Write(Utf8JsonWriter writer, Uri value, JsonSerializerOptio
         {
             writer.WriteStringValue(value.OriginalString);
         }
+
+        internal override Uri ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            Debug.Assert(reader.TokenType is JsonTokenType.PropertyName);
```

Comment:
was this throwing before?

Reply:
Yes, Uri key serialization was not supported.